modules/disk-uploader/pkg/secrets/secrets_test.go

```diff
@@ -0,0 +1,156 @@
+package secrets_test
+
+import (
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek8sclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/disk-uploader/pkg/secrets"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("VMExport", func() {
+	var (
+		kubeClient *fakek8sclient.Clientset
+	)
+
+	const (
+		namespace = "test-namespace"
+		name      = "test-vmexport"
+	)
+
+	BeforeEach(func() {
+		kubeClient = fakek8sclient.NewSimpleClientset()
+	})
+
+	Describe("CreateVirtualMachineExportSecret", func() {
+		Context("when the secret created successfully", func() {
+			BeforeEach(func() {
+				kubeClient.Fake.PrependReactor("get", "pods", func(action testing.Action) (bool, runtime.Object, error) {
+					_, ok := action.(testing.GetAction)
+					Expect(ok).To(BeTrue())
+
+					pod := &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      name,
+							Namespace: namespace,
+						},
+					}
+					return true, pod, nil
+				})
+				kubeClient.Fake.PrependReactor("create", "secrets", func(action testing.Action) (bool, runtime.Object, error) {
+					create, ok := action.(testing.CreateAction)
+					Expect(ok).To(BeTrue())
+
+					secret, ok := create.GetObject().(*corev1.Secret)
+					Expect(ok).To(BeTrue())
+					return true, secret, nil
+				})
+
+				os.Setenv("POD_NAME", name)
+				os.Setenv("POD_NAMESPACE", namespace)
```

Comment:
These same two env variables are set in two `Contexts()`, can you move them to a higher `Context()` and deduplicate?